source/common/router/config_impl.h

```diff
@@ -1029,7 +1044,8 @@ class ConfigImpl : public Config {
 public:
   ConfigImpl(const envoy::config::route::v3::RouteConfiguration& config,
              Server::Configuration::ServerFactoryContext& factory_context,
-             ProtobufMessage::ValidationVisitor& validator, bool validate_clusters_default);
+             ProtobufMessage::ValidationVisitor& validator, bool validate_clusters_default,
+             RdsStats* rds_stats = nullptr);
```

Comment:
Why is `rds_stats` a `nullptr` by default? Can it ever be a nullptr?

Reply:
For the bootstrap config, there will be nullptr passed in, since there is no rds_stats for bootstrap config.